src/LuaTransformer.ts

```diff
@@ -3666,6 +3670,91 @@ export class LuaTransformer {
         }
     }
 
+    public transformConsoleCallExpression(expression: ts.CallExpression): ExpressionVisitResult {
+        const method = expression.expression as ts.PropertyAccessExpression;
+        const methodName = method.name.escapedText;
+
+        switch (methodName) {
+            case "log":
+                if (expression.arguments.length > 0) {
+                    if (this.isStringFormatTemplate(expression.arguments[0])) {
+                        // print(string.format([arguments]))
+                        return tstl.createCallExpression(
+                            tstl.createIdentifier("print"),
+                            [tstl.createCallExpression(
+                                tstl.createTableIndexExpression(
+                                    tstl.createIdentifier("string"),
+                                    tstl.createStringLiteral("format")),
+                                this.transformArguments(expression.arguments))]
+                        );
+                    }
+                }
+                // print([arguments])
+                return tstl.createCallExpression(
+                    tstl.createIdentifier("print"),
+                    this.transformArguments(expression.arguments)
+                );
+            case "assert":
+                const args = this.transformArguments(expression.arguments);
+                if (expression.arguments.length > 1) {
+                    if (this.isStringFormatTemplate(expression.arguments[1])) {
+                        // assert([condition], string.format([arguments]))
+                        return tstl.createCallExpression(
+                            tstl.createIdentifier("assert"),
+                            [args[0],
+                            tstl.createCallExpression(
+                                tstl.createTableIndexExpression(
+                                    tstl.createIdentifier("string"),
+                                    tstl.createStringLiteral("format")),
+                                args.slice(1))]
+                        );
+                    }
+                }
+                // assert()
+                return tstl.createCallExpression(
+                    tstl.createIdentifier("assert"),
+                    args
+                );
+            case "trace":
+                if (expression.arguments.length > 0) {
+                    if (this.isStringFormatTemplate(expression.arguments[0])) {
```

Comment:
Might as well merge these two if statements into one with a `&&` (JS short-circuits if the left hand side fails, so it functions exactly the same)